libazureinit/Cargo.toml

```diff
@@ -12,7 +12,8 @@ description = "A common library for provisioning Linux VMs on Azure."
 reqwest = { version = "0.12.0", default-features = false, features = ["blocking", "json"] }
 serde = {version = "1.0.163", features = ["derive"]}
 thiserror = "1.0.58"
-tokio = { version = "1", features = ["full"] }
+# test-util feature is needed for time::advance in unit tests
+tokio = { version = "1", features = ["full", "test-util"] }
```

Comment:
You should be able to put this line in `[dev-dependencies]` and leave it as-is here so it's only enabled for tests.